### DISCUSSION TITLE
Systematic energy offsets between NN and DFT results

### DISCUSSION THREAD

**Discussion by LiangMD-BGI** [deepmodeling/deepmd-kit#514] [Q&A]
![Picture1](https://user-images.githubusercontent.com/66183652/114716034-f426ab80-9d33-11eb-8290-17a19ff218e5.png)
Hello all,

While forces are consistent between results from NN potential and DFT, 
there is a systematic offset of energy between NN and DFT results.

Below is my input.json file for deepmd: 

            "type": "se_a",
            "sel": [
                100,
                100
            ],
            "rcut_smth": 0.5,
            "rcut": 10.0,
            "neuron": [
                25,
                50,
                100
            ],
            "axis_neuron": 12,
            "seed": 3246556277
        },
        "fitting_net": {
            "neuron": [
                100,
                100,
                100
            ],
            "resnet_dt": true,
            "seed": 1978363284
        }
    },
    "learning_rate": {
        "start_lr": 0.001,
        "decay_steps": 5000,
        "decay_rate": 0.95,
        "_comment": " that's all"
    },
    "loss": {
        "start_pref_e": 1,
        "limit_pref_e": 1,
        "start_pref_f": 1000,
        "limit_pref_f": 100,
        "start_pref_v": 1,
        "limit_pref_v": 1
    },
    "training": {
        "systems": [
            "../data/"
        ],
        "set_prefix": "set",
        "stop_batch": 20000000,
        "batch_size": [
            1
        ],
        "disp_file": "lcurve.out",
        "disp_freq": 100,
        "numb_test": 4,
        "save_freq": 1000,
        "save_ckpt": "model.ckpt",
        "load_ckpt": "model.ckpt",
        "disp_training": true,
        "time_training": true,
        "profiling": false,
        "profiling_file": "timeline.json",
        "seed": 2662107775
    }
}

**Comment by LiangMD-BGI** [comment]
more information on lcurve file:

batch      l2_tst    l2_trn    l2_e_tst  l2_e_trn    l2_f_tst  l2_f_trn    l2_v_tst  l2_v_trn         lr
first line: 0    1.34e+02  1.31e+02    8.30e+00  8.62e+00    1.64e+00  1.04e+00    1.49e+00  2.19e-01    1.0e-03
latest line: 1153400    4.05e-01  4.51e-01    9.16e-03  9.63e-03    1.52e-02  1.80e-02    8.32e-03  8.53e-04    5.2e-04

**Reply by njzjz** [reply]
<s>l2_e_tst: 4.51e-01
l2_e_trn: 9.16e-03</s>
Can you explain what's the difference between your training data and test data? Also, which one do you plot, training or testing?

**Reply by LiangMD-BGI** [reply]
Hi Jinzhe,

Thanks for your reply.

It is 9.16e-03 vs. 9.63e-03 for energy in lcurve that I showed here (4.51e-01 is the loss function).
The data used for plotting are from "dp test" command, and test data is the whole data (that is "../data/").
Is there any setting in input.json file that looks unreasonable to your eyes?

With best regards,
Liang

**Reply by njzjz** [reply]
Got it. How many atoms are there in a cell? What’s the size of a cell?

**Reply by LiangMD-BGI** [reply]
I have 216 atoms. So the error of energy normalized by the number of atoms is about 0.005 eV/atom. 
I think this energy error is not large. But the problem is that the NN energy is systematically larger than the DFT energy.
Ideally one would expect that there are both positive and negative errors equally present.

**Reply by LiangMD-BGI** [reply]
The cell length is ~11.4 Ångström. and the pressure and temperature are 100 GPa and 2000 K.
The material is MgO (periclase).

**Reply by njzjz** [reply]
I guess the difference is the long-range interaction beyond the cutoff radius which cannot be trained. It’s the systematic error. (Not sure)

**Reply by finthon** [reply]
I also found the same problem, there is a systematic offset of energy between NN and DFT results, How did you fix this issue finally? @LiangMD-BGI

**Reply by LiangMD-BGI** [reply]
Hi, I have not figured out this problem. I also found that, during training, if you freeze the model at different times, the predicted NN energy could be either systematically larger or smaller than the one from DFT.  So I guess this error is related to the internal error of NN fitting. Looking forwards to having more discussion with you.

**Reply by njzjz** [reply]
I notice you set both `start_pref_e` and `limit_pref_e` to 1. You may consider to increase them although it may decrease the accuracy of forces. (Here energies and forces are inconsistent)

**Reply by LiangMD-BGI** [reply]
> I notice you set both `start_pref_e` and `limit_pref_e` to 1. You may consider to increase them although it may decrease the accuracy of forces. (Here energies and forces are inconsistent)

Hi Jinzhe, thanks for your suggestions. I will try it and see whether the results get better.

**Reply by Hashan-Peiris** [reply]
I am facing the same issue. Was it fixed by any chance?

**Comment by J-You** [comment]
I want to know if it is  suitable to set "rcut_smooth" as 0.5 in your input.json file?
Maybe it is equal to 9.5 according to your "rcut" setting?

**Reply by LiangMD-BGI** [reply]
The rcut_smth gives where the smoothing starts. The setting (rcut_smth = 0.5 and rcut = 10), therefore, allows smoothing in a larger range of radius relative to the one you suggested (rcut_smth = 9.5 and rcut = 10). 
Such a small rcut_smth = 0.5 is present in several works (e.g., https://www.pnas.org/content/117/42/26040.short).
So there should be no problems there.

**Reply by J-You** [reply]
Thanks for your response. It extends my understanding scope.

**Comment by mirza-galib** [comment]
Hi, I am not sure if the offset between DFT energy and NN energy problem is solved. You are getting this because your weight factor on the force is high and on the energy is low. You need to end with equal weight on force and energy to improve the accuracy on energy training.

**Reply by LiangMD-BGI** [reply]
Hi, thank you very much for your suggestion.  I will try it and see whether the results get better.

**Comment by sooyaaa233** [comment]
Did you solve your problem now? How did you solve it?
